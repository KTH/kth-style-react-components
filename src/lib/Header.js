import React from 'react'
import cortina from 'kth-node-cortina-block'

class Header extends React.Component {
  constructor() {
    super()
    this.state = {
      secondaryMenu: '',
      megaMenu: '',
      image: '',
      title: '',
      search: ''
    }
  }

  componentDidUpdate() {
    this.componentDidMount()
  }

  componentDidMount() {
    cortina({ url: this.props.url, language: this.props.language }).then(blocks => {
      return this.setState({
        secondaryMenu: blocks.secondaryMenu,
        megaMenu: blocks.megaMenu,
        image: blocks.image,
        title: this.modifyTitle(blocks.title, this.props.appTitle, this.props.appLink),
        search: blocks.search
      })
    })
  }

  modifyTitle(cortinaBlock, appTitle, appLink) {
    if (appTitle) {
      return cortinaBlock.replace(/href=[^.]+<\/a>/, `href="${appLink}">${appTitle}</a>`)
    }
    return cortinaBlock
  }

  concatinateHTML(section) {
    if (section === 'top') {
      return this.state.image + this.state.title + this.state.secondaryMenu
    } else if (section === 'bottom') {
      return this.state.megaMenu + this.state.search
    } else {
      return ''
    }
  }

  render() {
    return (
      <header role="banner">
        <div className="container-fluid">
          <div className="container">
            <div
              className="header-container__top"
              dangerouslySetInnerHTML={{ __html: this.concatinateHTML('top') }}
            />
            <div
              className="header-container__bottom"
              dangerouslySetInnerHTML={{ __html: this.concatinateHTML('bottom') }}
            />
          </div>
        </div>
        <div id="gradientBorder" />
      </header>
    )
  }
}

Header.defaultProps = {
  url: 'https://www.kth.se/cm/',
  language: 'sv',
  appTitle: '',
  appLink: '/'
}

export default Header
