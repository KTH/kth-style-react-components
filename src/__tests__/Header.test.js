import React from "react";
import renderer from "react-test-renderer";
import Header from '../lib/Header';

describe("Header", () => {

  // TODO: await cortina api call
  it("should match previous snapshot", async () => {
    const tree = renderer
      .create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});