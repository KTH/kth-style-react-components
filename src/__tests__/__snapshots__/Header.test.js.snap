// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match previous snapshot 1`] = `
<header
  role="banner"
>
  <div
    className="container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="header-container__top"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        className="header-container__bottom"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
  <div
    id="gradientBorder"
  />
</header>
`;
